docs/architecture-decisions/multiple-sync-sources.md

```diff
@@ -0,0 +1,82 @@
+---
+# Valid statuses: draft | proposed | rejected | accepted | superseded
+status: draft
+author: Dave Josephsen
+created: 2025-05-21
+updated: 2025-05-21
+---
+
+# ADR: Multiple Sync Sources
+It is the Intent of this document to articulate our rationale for supporting multiple flag synchronization sources (grpc, http, blob, local file, etc..) as a core design property of Flagd. Because "future proofing" the runtime against sources that don't yet exist, or those we may have omitted is a requisite byproduct of this architectural decision, this document also includes a short discussion of how flagd is engineered to enable the community to extend it to support new sources in the future.
+
+The goal of first-class multi-sync support generally is to broaden Flagd's potential to suit the needs of many different types of engineers, operating in many different contexts. By decoupling flag persistence from the runtime, Flagd can focus on evaluation and sync, while enabling its user-base to choose a persistence layer that best suits their individual requirements. 
+
+## Background
```

Comment:
Trying to understand more of the background - is this a backfill ADR? 
It also confuses me a bit as I don't see the connection between storage and multi sync sources - is the concern around the data consistency?